Adding External Logging Link for Containers Providers

enoodle opened the pull request:
This also adds the links for container nodes for when support for this feature will be added to container nodes.

![launch_kibana](https://cloud.githubusercontent.com/assets/3123328/23481740/34f61e68-fed5-11e6-9a15-72325478911f.png)

------------------------------

![no_route_for_external_logging](https://cloud.githubusercontent.com/assets/3123328/23481747/3d3cd3aa-fed5-11e6-8f40-0b3b72cb3c21.png)


--------------------------------

![gifrecord_2017-03-01_231329](https://cloud.githubusercontent.com/assets/3123328/23481755/435041c8-fed5-11e6-9bd6-7d59c3abf7f2.gif)


This is a "resubmission" of https://github.com/ManageIQ/manageiq-ui-classic/pull/36 due to Travis problems with the previous PR.

enoodle commented:
cc @simon3z

simon3z commented:
@enoodle can you add a screenshot of the error case "A route named '#{route_name}' is configured to connect to the..."

simon3z commented on app/controllers/mixins/containers_external_logging_support_mixin.rb:
```diff
@@ -0,0 +1,17 @@
+    route_name = ems.external_logging_route_name
+    logging_route = ContainerRoute.find_by(:name => route_name, :ems_id => ems.id) if route_name
+    if logging_route
+      url = "https://#{logging_route.host_name}#{record.external_logging_path}"
```
@enoodle to be precise also http/https is specified in the route (we should use what the route specifies).

enoodle replied:
@simon3z My routes don't have a http/https specified in them. I guess this is optional to them. I can check for it and use the route's protocol if it is mentioned.

enoodle replied:
@simon3z I updated to do that ^

simon3z commented:
A couple of open questions. Rest LGTM.
@dclarizio @h-kataria can you review?

@miq-bot assign dclarizio

dclarizio commented:
@enoodle @simon3z

I'm thinking the button shouldn't be to the left of Configuration, as it is pretty standard that Configuration is the left most main button.  Seeing as it has the same image as Monitoring, should it maybe go inside that button group?

Also, what happens when this button is pressed, is a screen presented?  Maybe it's something that already exists in the UI?

enoodle commented:
@simon3z the flash error will only show in the very rare occasion when the route was deleted after the UI wsa loaded and a refresh to the provider has seen that.  I added this error message to the button when disabling it and added a screenshot of it on the top comment.

@dclarizio This button will open a new window for the external logging. I moved it to the left of the monitoring button but I could put it inside if you would like it there. We could also ask for a new icon for logging. please advise.

dclarizio commented:
@enoodle 
>@dclarizio This button will open a new window for the external logging. I moved it to the left of the monitoring button but I could put it inside if you would like it there. We could also ask for a new icon for logging. please advise.

To me, if feels like it would fit under Monitoring, but perhaps that decision should come from the Containers team.  If we leave it as a main button, we should definitely have a different image.  If needed, we can have @epwinchell take care of that.

@simon3z can you help with this decision?  Thx, Dan

Loicavenel commented:
@dclarizio It is provider related, each provider has his own log server. today It is only for OpenShift Provider, in the future, other Red Hat providers will potentially use it. Also, we will go down to lower object level and offering it for Project, Image etc..

simon3z commented:
> @simon3z can you help with this decision? Thx, Dan

@dclarizio actually I had the same thought: indeed it could fit under Monitoring.
On the other hand the VM console (from which we took inspiration as external link) is a separate button.

So I have no preferred solution, in a way or another, both make sense for me.

Loicavenel commented:
@dclarizio I did not understand your question and reading Federico, this is making sense now. I think under monitor "Menu" will make more sense, don't need to add another menu.

dclarizio commented:
@enoodle after you address the above comments, please revise the screen shots and we can get this merged.  Thx, Dan

enoodle commented:
@dclarizio PTAL

dclarizio commented:
@enoodle 👍 merging!
@epwinchell can you suggest another image for the new button, in another PR?  Thx, Dan
